cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -853,19 +854,21 @@ public void searchMetadata(HttpRequest request, HttpResponder responder,
 
   private Set<MetadataRecord> getMetadata(Id.NamespacedId entityId,
                                           @Nullable String scope) throws NotFoundException, BadRequestException {
-    return  (scope == null) ? metadataAdmin.getMetadata(entityId) :
-      metadataAdmin.getMetadata(validateScope(scope), entityId);
+    return  (scope == null) ? metadataAdmin.getMetadata((NamespacedId) entityId.toEntityId()) :
+      metadataAdmin.getMetadata(validateScope(scope), (NamespacedId) entityId.toEntityId());
   }
 
   private Map<String, String> getProperties(Id.NamespacedId entityId,
```

Comment:
Also, make this method accept ``NamespacedId``, so you don't need casting below.